brvs-core/src/main/java/iroha/validation/transactions/provider/impl/AccountManager.java

```diff
@@ -246,26 +250,37 @@ public void register(String accountId) throws InterruptedException {
    * {@inheritDoc}
    */
   @Override
-  public void register(Iterable<String> accounts) throws InterruptedException {
-    final int size = Iterables.size(accounts);
-    final RegistrationAwaiterWrapper registrationAwaiterWrapper = new RegistrationAwaiterWrapper(
-        new CountDownLatch(size)
-    );
-
-    accounts.forEach(account -> executorService
-        .submit(new RegistrationRunnable(account, registrationAwaiterWrapper))
-    );
+  public void register(Collection<String> accounts) throws InterruptedException {
+    final List<List<String>> partitions = Lists
+        .partition(
+            new ArrayList<>(accounts),
+            REGISTRATION_BATCH_SIZE
+        );
+    for (List<String> partition : partitions) {
+      final int size = Iterables.size(partition);
+      final RegistrationAwaiterWrapper registrationAwaiterWrapper = new RegistrationAwaiterWrapper(
+          new CountDownLatch(size)
+      );
 
-    if (!registrationAwaiterWrapper.getCountDownLatch().await(size * 10L, TimeUnit.SECONDS)) {
-      throw new BrvsException("Couldn't register accounts within a timeout", REGISTRATION_TIMEOUT);
-    }
-    final Exception registrationAwaiterWrapperException = registrationAwaiterWrapper.getException();
-    if (registrationAwaiterWrapperException != null) {
-      throw new BrvsException(
-          registrationAwaiterWrapperException.getMessage(),
-          registrationAwaiterWrapperException,
-          REGISTRATION_FAILED
+      partition.forEach(account -> executorService
```

Comment:
I believe that a fixed pool of threads, in which the number of workers is equal to the number of available cores, will provide better performance